### PR TITLE
Exempt certain methods from call target trimming

### PR DESCRIPTION
**Pull request by jdmpapin**
See the commit messages for more detail

### CODE REVIEW

**Comment by jdmpapin** [comment]
@0xdaryl, @vijaysun-omr

File: runtime/compiler/optimizer/InlinerTempForJ9.cpp [hunk 1]
```diff
@@ -4012,8 +4012,10 @@ bool TR_MultipleCallTargetInliner::inlineSubCallGraph(TR_CallTarget* calltarget)
     * 1. It's a JSR292 related method. This condition allows inlining method handle thunk chain without inlining the leaf java method.
     * 2. It's force inline target
     */
+   TR::Node *callNode = NULL; // no call node has been generated yet
```
Comment (vijaysun-omr) [inline runtime/compiler/optimizer/InlinerTempForJ9.cpp:4015]: Should the comment above this code be updated to reflect the 3rd case that has been added now ?

**Review by vijaysun-omr — APPROVED** [review]

File: runtime/compiler/optimizer/InlinerTempForJ9.cpp (hunk 1 above)
Comment (jdmpapin) [reply runtime/compiler/optimizer/InlinerTempForJ9.cpp:4015]: Sure, I'll update the comment

**Comment by jdmpapin** [comment]
Pushed an update to the comment

**Comment by vijaysun-omr** [comment]
Jenkins test extended all jdk8,jdk11,jdk16

**Comment by vijaysun-omr** [comment]
Jenkins test extended xlinux,zlinux jdk16

**Comment by vijaysun-omr** [comment]
Jenkins test extended xlinux,zlinux jdk11

**Comment by 0xdaryl** [comment]
Jenkins test extended osx jdk11

macOS failure appears infrastructural.

**Comment by vijaysun-omr** [comment]
Checks have all passed and review is done. Merging.

**Pull request merged**